esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -4202,20 +4206,27 @@ void CWsWuFileHelper::sendWUComponentLogStreaming(CHttpRequest* request, CHttpRe
     }
 }
 
-void CWsWuFileHelper::readWUComponentLogOptionsReq(IConstWUFileOption& logOptionReq, WUComponentLogOptions& options)
+void CWsWuFileHelper::readWUComponentLogOptionsReq(CWULogFileRequest* logReq, IConstWUFileOption* logOptionReq, WUComponentLogOptions& options)
 {
-    //If MaxLogRecords is in logOptionReq, use it; otherwise, default to 100 in LogAccessConditions.
-    if (!logOptionReq.getMaxLogRecords_isNull())
-        options.logFetchOptions.setLimit(logOptionReq.getMaxLogRecords());
-    //If LogSearchTimeBuffSecs is in logOptionReq, use it; otherwise, default to defaultWULogSearchTimeBufferSecs.
-    if (!logOptionReq.getLogSearchTimeBuffSecs_isNull())
-        options.wuLogSearchTimeBuffSecs = logOptionReq.getLogSearchTimeBuffSecs();
-    CLogAccessLogFormat logFormatSetting = logOptionReq.getLogFormat();
+    //If MaxLogRecords is in logReq, use it; otherwise, default to 100 in LogAccessConditions.
```

Comment:
I think it would be more intuitive/readable if this was organized into two sections...

```
if (logReq)
{
}
else if (logOptionReq)
{
}
```